src/main/java/org/javarosa/core/model/ItemsetBinding.java

```diff
@@ -197,7 +210,8 @@ private SelectChoice getChoiceForTreeReference(FormDef formDef, DataInstance for
             labelIsItext,
             item.getInstanceName() != null
                 ? (TreeElement) formDef.getNonMainInstance(item.getInstanceName()).resolveReference(item)
-                : formDef.getMainInstance().resolveReference(item));
+                : formDef.getMainInstance().resolveReference(item),
+            labelRef.getNameLast());
```

Comment:
This won't work in the case of a nested reference which I believe would be possible when using an XML secondary instance. That would look like:

```
        <itemset nodeset="instance('cities')/root/item[country= /data/loc/country ]">
            <value ref="name"/>
            <label ref=➡️"basic_deets/human_readable_stuff/label"⬅️/>   
        </itemset>
```

I've never seen it in use. I think we should start with this and handle any issues users bring up.